### ISSUE TITLE
regenerate.sh should not require pygraphviz to be installed to run

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#17336]
```
(/Users/ezyang/Dev/pytorch-tmp-env) macbook-pro-116:.circleci ezyang$ ./regenerate.sh 
+ ./generate-config-yml.py
Traceback (most recent call last):
  File "./generate-config-yml.py", line 14, in <module>
    import binary_build_definitions
  File "/Users/ezyang/Dev/pytorch-tmp/.circleci/binary_build_definitions.py", line 3, in <module>
    import conf_tree
  File "/Users/ezyang/Dev/pytorch-tmp/.circleci/conf_tree.py", line 4, in <module>
    from pygraphviz import AGraph
ModuleNotFoundError: No module named 'pygraphviz'
```

**Comment by kostmo** [comment]
Fixed in commit dd3acbc6d

**Comment by kostmo** [comment]
actually this still needs some work:
```
Traceback (most recent call last):
  File "./generate_config_yml.py", line 105, in <module>
    stitch_sources(sys.stdout)
  File "./generate_config_yml.py", line 100, in stitch_sources
    f.write(output_filehandle)
  File "./generate_config_yml.py", line 42, in write
    self.function(build_dict)
  File "/home/kostmo/github/pytorch-repos/pytorch/.circleci/cimodel/pytorch_build_definitions.py", line 409, in add_build_env_defs
    graph = visualization.generate_graph(get_root())
  File "/home/kostmo/github/pytorch-repos/pytorch/.circleci/cimodel/visualization.py", line 30, in generate_graph
    from pygraphviz import AGraph
ModuleNotFoundError: No module named 'pygraphviz'
```
